experimental/framework/fluid-static/src/containerCode.ts

```
@@ -115,6 +122,25 @@ export class RootDataObject
         super.dispose();
     }
 
+    public setContainer(container: Container) {
+        this._container = container;
+    }
```

Let's add a check and throw if container is not undefined to prevent re-entry.